packages/orm/src/EntityManager.ts

```diff
@@ -522,12 +556,25 @@ export class EntityManager {
     const joinRowTodos = sortJoinRows(this.__data.joinRows);
 
     if (Object.keys(entityTodos).length > 0 || Object.keys(joinRowTodos).length > 0) {
-      await this.knex.transaction(async (tx) => {
-        await flushEntities(this.knex, tx, entityTodos);
-        await flushJoinTables(this.knex, tx, joinRowTodos);
-        await tx.commit();
-      });
+      const alreadyInTxn = "commit" in this.knex;
+
+      if (!alreadyInTxn) {
+        await this.knex.transaction(async (knex) => {
```

Comment:
if we changed `em.transaction` so that nested calls just run their callback without spawning a new transaction, then this code could be simplified to remove the check for a transaction.